src/serialize.h

```diff
@@ -165,11 +165,10 @@ enum
 };
 
 //! Convert the reference base type to X, without changing constness or reference type.
-template<typename X> X& ReadWriteAsHelper(X& x) { return x; }
-template<typename X> const X& ReadWriteAsHelper(const X& x) { return x; }
+template<typename X> X& AsBase(X& x) { return x; }
```

Comment:
It needs to maintain constness. During serialization, the object may be `const T`, while during deserialization it cannot be `const`. This function is effectively a static cast, but one that passes through constness.

Reply:
Added more comments to clarify this.

Reply:
I checked that replacing `s/AsBase/static_cast/` all over the place compiles and the few tests I ran pass.

Reply:
Yes, that's expected - but it may lead to UB. `static_cast` may cast away the constness of a reference, in a place where it is illegal to do so. So `AsBase` is just a more restrictive, safer, version of `static_cast`.